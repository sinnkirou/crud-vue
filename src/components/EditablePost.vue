<template>
  <div class="post" key="{{`EditablePost${post.id}`}}">
    <form class="form" @submit.prevent="onSubmit">
      <input required type="text" placeholder="Enter Post Title" v-model="formObj.title">
      <br>
      <br>
      <textarea required rows="5" cols="28" placeholder="Enter Post" v-model="formObj.message"/>
      <br>
      <br>
      <button
        class="mdl-button mdl-js-button mdl-button--raised"
        type="submit"
      >{{post&&post.id ? "Update": "Create"}}</button>
    </form>
  </div>
</template>

<script>
export default {
  name: "EditablePost",
  props: {
    post: Object
  },
  data: function() {
    return {
      formObj: {
        ...this.post
      }
    };
  },
  inject: ["addPost", "updatePost"],
  methods: {
    reset: function() {
      this.formObj.title = "";
      this.formObj.message = "";
    },
    onSubmit: function() {
      if (!this.post || !this.post.id) {
        this.addPost(this.formObj);
        this.reset();
      } else {
        this.updatePost(this.formObj);
        this.$emit("toggle-Editable");
      }
    }
  }
};
</script>
