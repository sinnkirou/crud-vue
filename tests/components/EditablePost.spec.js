import { shallowMount } from "@vue/test-utils";
import EditablePost from "@/components/EditablePost.vue";

describe("EditablePost.vue", () => {
  let wrapper;
  const post = {
    id: "1",
    title: "title",
    message: "message"
  };

  beforeEach(() => {
    wrapper = shallowMount(EditablePost, {
      propsData: { post }
    });
  });

  it("should render", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper.vm.post).toEqual(post);
  });

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
