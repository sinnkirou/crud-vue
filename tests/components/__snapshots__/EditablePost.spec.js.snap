// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditablePost.vue has the expected html structure 1`] = `
<div
  class="post"
>
  <form
    class="form"
  >
    <input
      placeholder="Enter Post Title"
      required="required"
      type="text"
    />
     
    <br />
     
    <br />
     
    <textarea
      cols="28"
      placeholder="Enter Post"
      required="required"
      rows="5"
    />
     
    <br />
     
    <br />
     
    <button
      class="mdl-button mdl-js-button mdl-button--raised"
      type="submit"
    >
      Update
    </button>
  </form>
</div>
`;
